src/pdf_parser/pdf_utils/disambiguate_layout.py

```diff
@@ -332,23 +332,28 @@ def unnest_boxes(layout: Layout, unnest_soft_margin: int = 15) -> Layout:
 
 # TODO: This is not part of the module and is for a CLI, but placing here for visibility before I edit the CLI.
 def disambiguation_pipeline(
```

Comment:
Small comment but could this be `run_disambiguation_pipeline`? I find that starting functions with a verb helps with understanding what they do, and allows adding other methods to do with the pipeline later e.g. `configure_disambiguation_pipeline`.

Also, I'm not immediately clear what this does from looking at the function. Could you add a line to the docstring explaining what you mean by disambiguation?